recipes/gtk/all/conanfile.py

```diff
@@ -0,0 +1,156 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibnameConan(ConanFile):
+    name = "gtk"
+    description = "libraries used for creating graphical user interfaces for applications."
+    topics = ("conan", "gtk", "widgets")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gtk.org"
+    license = "LGPL-2.1-or-later"
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_wayland": [True, False],
+        "with_x11": [True, False],
+        "with_pango": [True, False]
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_wayland": False,
+        "with_x11": True,
+        "with_pango": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.with_wayland
+            del self.options.with_x11
+
+    def validate(self):
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+            raise ConanInvalidConfiguration("this recipes does not support GCC before version 5. contributions are welcome")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Linux":
+            if self.options.with_wayland or self.options.with_x11:
+                if not self.options.with_pango:
+                    raise ConanInvalidConfiguration("with_pango option is mandatory when with_wayland or with_x11 is used")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("GTK recipe is not yet compatible with Windows. Contributions are welcome.")
```

Comment:
What are the problems with Windows?

Reply:
the first blocking one is that ninja fails unless you use env var `CONAN_USE_ALWAYS_SHORT_PATHS=1` (`short_paths` attribute is not enough)